File: src/test/java/com/medicalsystem/repository/FieldRepositoryTest.java
package com.medicalsystem.repository;

import com.medicalsystem.model.Field;
import com.medicalsystem.model.FieldType;
import com.medicalsystem.util.GenUtils;
import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.List;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

@RunWith(SpringRunner.class)
@SpringBootTest
public class FieldRepositoryTest {

    @Autowired
    private FieldRepository fieldRepository;

    @Test
    public void test_autowire() {
        assertNotNull(fieldRepository);
    }

    @Test
    public void test_save_and_findOne() {
        Field field = GenUtils.getRandomField();

        field = fieldRepository.save(field);
        field = fieldRepository.findOne(field.getId());

        assertNotNull(field);
    }

    @Test
    public void test_findAll() {
        final long fieldCountBefore = fieldRepository.count();
        final int fieldsToSave = 10;

        for (int i = 0; i < fieldsToSave; i++) {
            Field field = GenUtils.getRandomField();
            fieldRepository.save(field);
        }

        List<Field> fields = fieldRepository.findAll();

        assertNotNull(fields);
        assertEquals(fieldsToSave + fieldCountBefore, fields.size());
    }

}

File: src/test/kotlin/com/medicalsystem/repository/FieldRepositoryTest.kt
package com.medicalsystem.repository

import com.medicalsystem.model.Field
import com.medicalsystem.model.FieldType
import com.medicalsystem.model.Form
import com.medicalsystem.model.Section
import org.junit.After
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit4.SpringRunner

@RunWith(SpringRunner::class)
@SpringBootTest
class FieldRepositoryTest {

    @Autowired lateinit var fieldRepository: FieldRepository
    @Autowired lateinit var formRepository: FormRepository

    @Before @After
    fun tearDown() {
        formRepository.deleteAll()
        fieldRepository.deleteAll()
    }

    @Test
    fun testAutowire() {
        Assert.assertNotNull(fieldRepository)
    }

    @Test
    fun testSave() {
        val field = Field(name = "Pole 1", type = FieldType.SELECT)
        val saved = fieldRepository.save(field)
        Assert.assertTrue(saved.id > 0)
    }

    @Test
    fun testFindAllByForm() {
        val form = Form()
        val sections: List<Section> = listOf(Section(), Section(), Section())
        val fields: List<List<Field>> = listOf(
                listOf(Field(), Field(), Field()),
                listOf(Field(), Field(), Field()),
                listOf(Field(), Field(), Field())
        )

        form.sections = sections.toMutableList()
        sections.forEach { it.form = form }

        sections.forEachIndexed { index, section -> section.fields = fields[index].toMutableList() }
        fields.forEachIndexed { index, list -> list.forEach { it.section = sections[index] } }

        val savedForm = formRepository.save(form)

        Assert.assertEquals(9, fieldRepository.findAllByForm(savedForm).size)
    }
}
